File: Code/Cplus/LAB07_02/LAB07_02.cpp
#include <iostream>
using namespace std;

// ฟังก์ชันสำหรับหาค่ามากที่สุดในอาร์เรย์
int findMax(int arr[], int size) {
    int maxVal = arr[0];
    for (int i = 1; i < size; i++) {
        if (arr[i] > maxVal) {
            maxVal = arr[i];
        }
    }
    return maxVal;
}

// ฟังก์ชันสำหรับหาค่าน้อยที่สุดในอาร์เรย์
int findMin(int arr[], int size) {
    int minVal = arr[0];
    for (int i = 1; i < size; i++) {
        if (arr[i] < minVal) {
            minVal = arr[i];
        }
    }
    return minVal;
}

int main() {
    int size;

    // รับขนาดของอาร์เรย์จากผู้ใช้
    cout << "Enter the size of the array: ";
    cin >> size;

    // สร้างอาร์เรย์แบบไดนามิก
    int* arr = new int[size];

    // รับค่าของอาร์เรย์
    cout << "Enter " << size << " elements:" << endl;
    for (int i = 0; i < size; i++) {
        cin >> arr[i];
    }

    // เรียกฟังก์ชันเพื่อหาค่ามากที่สุดและน้อยที่สุด
    int maxVal = findMax(arr, size);
    int minVal = findMin(arr, size);

    // แสดงผลลัพธ์
    cout << "\nMaximum value in the array: " << maxVal << endl;
    cout << "Minimum value in the array: " << minVal << endl;

    // คืนหน่วยความจำที่จัดสรร
    delete[] arr;

    return 0;
}

File: Code/Python/LAB07_02.py
def find_max(arr):
    """ฟังก์ชันสำหรับหาค่ามากที่สุดในอาร์เรย์"""
    max_val = arr[0]
    for num in arr:
        if num > max_val:
            max_val = num
    return max_val

def find_min(arr):
    """ฟังก์ชันสำหรับหาค่าน้อยที่สุดในอาร์เรย์"""
    min_val = arr[0]
    for num in arr:
        if num < min_val:
            min_val = num
    return min_val

def main():
    # รับขนาดของอาร์เรย์จากผู้ใช้
    size = int(input("Enter the size of the array: "))

    # รับค่าของอาร์เรย์
    arr = []
    print(f"Enter {size} elements:")
    for _ in range(size):
        num = int(input())
        arr.append(num)

    # หาค่ามากที่สุดและน้อยที่สุด
    max_val = find_max(arr)
    min_val = find_min(arr)

    # แสดงผลลัพธ์
    print(f"\nMaximum value in the array: {max_val}")
    print(f"Minimum value in the array: {min_val}")

if __name__ == "__main__":
    main()
